vmware-tanzu/nsx-operator
# Remove NetworkInfo Finalizer

This PR attempts to resolve the issue of a Namespace being stuck in the terminating state. Due to the constraints of the Finalizer mechanism, when a Namespace is created and then deleted, then if we stop the Pod or other abnormal conditions in the Pod, the NetworkInfo resource cannot be cleaned up, causing the Namespace to remain stuck in the terminating state. This PR removes the Finalizer mechanism for deleting NetworkInfo.

Also removed Subnet/SubnetSet Finalizer in https://github.com/vmware-tanzu/nsx-operator/pull/769

TestDone:

1. Create a Namespace and check the Networkinfo status; 
2. Delete the Namespace and check for any stale VPC in NSX.

Rebase https://github.com/vmware-tanzu/nsx-operator/pull/769 and test the deletion of a Namespace, then delete the Pod, the Namespace will not stuck in terminating status.
